azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/EventProcessorHost.java

```diff
@@ -28,7 +28,7 @@
     private final int executorServicePoolSize = 16;
     private final HostContext hostContext;
     private boolean initializeLeaseManager = false;
-    private boolean unregistered = false;
+    private CompletableFuture<Void> unregistered = null;
```

Comment:
mark volatile
to cover the case when register and unregister are called from different threads..